python/box.tests/test/org/jetbrains/kotlin/python/test/BasicIrBoxTest.kt

```diff
@@ -138,23 +142,37 @@ abstract class BasicIrBoxTest(
             }
 
             if (!skipRegularMode) {
-                val compiledModule = compile(
-                    project = config.project,
-                    mainModule = MainModule.SourceFiles(filesToCompile),
-                    analyzer = AnalyzerWithCompilerReport(config.configuration),
-                    configuration = config.configuration,
-                    phaseConfig = phaseConfig,
-                    irFactory = IrFactoryImpl,
-                    allDependencies = resolvedLibraries,
-                    friendDependencies = emptyList(),
-                    mainArguments = mainCallParameters.run { if (shouldBeGenerated()) arguments() else null },
-                    exportedDeclarations = setOf(FqName.fromSegments(listOfNotNull(testPackage, testFunction))),
-                    generateFullJs = true,
-                    generateDceJs = runIrDce,
-                    es6mode = runEs6Mode,
-                    multiModule = splitPerModule || perModule,
-                    propertyLazyInitialization = propertyLazyInitialization,
-                )
+                var compilationException: Throwable? = null
+                val compiledModuleWithDuration: Pair<Long, CompilerResult?> = measureTimeMillisWithResult {
+                    try {
+                        compile(
+                            project = config.project,
+                            mainModule = MainModule.SourceFiles(filesToCompile),
+                            analyzer = AnalyzerWithCompilerReport(config.configuration),
+                            configuration = config.configuration,
+                            phaseConfig = phaseConfig,
+                            irFactory = IrFactoryImpl,
+                            allDependencies = resolvedLibraries,
+                            friendDependencies = emptyList(),
+                            mainArguments = mainCallParameters.run { if (shouldBeGenerated()) arguments() else null },
+                            exportedDeclarations = setOf(FqName.fromSegments(listOfNotNull(testPackage, testFunction))),
+                            generateFullJs = true,
+                            generateDceJs = runIrDce,
+                            es6mode = runEs6Mode,
+                            multiModule = splitPerModule || perModule,
+                            propertyLazyInitialization = propertyLazyInitialization,
+                        )
+                    } catch (e: Throwable) {
+                        compilationException = e
+                        null
+                    }
+                }
+                val compilationTimeMessage = "Kotlin compilation time: ${compiledModuleWithDuration.first} ms"
+
+                compilationException?.let {
+                    throw KotlinCompilationException(compilationTimeMessage, it)
+                }
+                val compiledModule = compiledModuleWithDuration.second!!
```

Comment:
Well, we can still improve code around here a bit, but it's very minor and can be addressed later if we want, so I don't mind to merge PR in the current state without further changes.

In the future, improvement can be like this:

```suggestion
                var compilationException: Throwable? = null
                val (duration, compilerResult) = measureTimeMillisWithResult {
                    try {
                        compile(
                            project = config.project,
                            mainModule = MainModule.SourceFiles(filesToCompile),
                            analyzer = AnalyzerWithCompilerReport(config.configuration),
                            configuration = config.configuration,
                            phaseConfig = phaseConfig,
                            irFactory = IrFactoryImpl,
                            allDependencies = resolvedLibraries,
                            friendDependencies = emptyList(),
                            mainArguments = mainCallParameters.run { if (shouldBeGenerated()) arguments() else null },
                            exportedDeclarations = setOf(FqName.fromSegments(listOfNotNull(testPackage, testFunction))),
                            generateFullJs = true,
                            generateDceJs = runIrDce,
                            es6mode = runEs6Mode,
                            multiModule = splitPerModule || perModule,
                            propertyLazyInitialization = propertyLazyInitialization,
                        )
                    } catch (e: Throwable) {
                        compilationException = e
                        null
                    }
                }

                compilationException?.let {
                    val compilationTimeMessage = "Kotlin compilation time: $duration ms"
                    throw KotlinCompilationException(compilationTimeMessage, it)
                }
                val compiledModule = compilerResult!!  // maybe we can get rid of !! too somehow
```